### PR TITLE
feat: protocol rewards

### PR DESCRIPTION
**Pull request by kulkarohan**

### CODE REVIEW

File: foundry.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [profile.default]
 solc_version = '0.8.17'
 optimizer = true
-optimizer_runs = 2383
+optimizer_runs = 500
```
Comment (iainnash) [inline foundry.toml:4]: :'(

File: foundry.toml [hunk 2]
```diff
@@ -21,4 +21,6 @@ optimism_goerli = "https://goerli.optimism.io"
 zora = "https://rpc.zora.energy"
 zora_goerli = "https://testnet.rpc.zora.energy"
 base_goerli = "https://goerli.base.org"
-base = "https://developer-access-mainnet.base.org"
\ No newline at end of file
+base = "https://developer-access-mainnet.base.org"
```
Comment (iainnash) [inline foundry.toml:24]: I believe this is here now – can you rebase?

File: script/TestUpgradeAndMint.s.sol [hunk 3]
```diff
@@ -22,6 +22,8 @@ import {ERC721DropProxy} from "../src/ERC721DropProxy.sol";
 contract DeployNewERC721Drop is Script {
     using Strings for uint256;
 
+    address internal DEFAULT_CREATE_REFERRAL = address(0);
```
Comment (iainnash) [inline script/TestUpgradeAndMint.s.sol:25]: should this be immutable?

Comment (kulkarohan) [reply script/TestUpgradeAndMint.s.sol]: removed, wasn't necessary

**Review by jgeary — APPROVED** [review]

**Pull request merged**